src/main/php/CommandLine/Library/FinderToPathsConverter.php

```diff
@@ -5,25 +5,35 @@
 use Symfony\Component\Finder\Finder;
 use Symfony\Component\Finder\SplFileInfo;
 
+/**
+ * This converter is able to extract data from a
+ * Symfony\Component\Finder\Finder object and returns them
+ * in a more useful data structure.
+ */
 class FinderToPathsConverter
 {
     /**
-     * Converts Finder objects to Arrays of their relative paths.
+     * Converts Finder objects to Arrays of their relative paths without the filename.
      *
      * @param Finder $finder
      *
      * @return string[]
      */
-    public function finderToArrayOfPaths(Finder $finder) : array
+    public function finderToArrayOfPaths(Finder $finder): array
     {
-        $result = array_map(
-            function ($value) {
-                /** @var SplFileInfo $value */
-                return $value->getRelativePathname();
+        $directories = array_map(
+            static function (SplFileInfo $value) {
+                if ($value->isDir()) {
+                    return $value->getRelativePathname() . '/';
+                }
+                if ($value->getRelativePath() === '') {
+                    return './';
+                }
+                return $value->getRelativePath() . '/';
```

Comment:
wird das letzte return gebraucht? kann ja kein directory mehr sein

Reply:
Die Methode soll ja nur relative Pfade zu Verzeichnissen zurück geben. Finder spukt 3 verschiedene Arten von SplFileInfos aus.

1) Zu Verzeichnissen (ohne trailing Slash) Hier müssen wir also den Namen nutzen.
  ```php
if ($value->isDir()) {
    return $value->getRelativePathname() . '/';
}
  ```
2) Einen Link zu einer Datei, die in keinem Verzeichnis liegt - deren relatives Verzeichnis also `''` wäre und damit aus Linuxsicht `./`
  ```php
if ($value->getRelativePath() === '') {
    return './';
}
  ```
3) Dateien, die in einem Unterverzeichnis liegen. Relative Links zu deren Verzeichnissen sind also nicht leer
  ```php
return $value->getRelativePath() . '/';
  ```

Reply:
ah, okay. getRelativePathname() vs. getRelativePath() hab ich übersehen. SplFileInfo hat nicht die bestbenamte API der welt.